File: frontend/src/types/verification.ts
export interface VerificationRead {
  id: string;
  company_id: string;
  company_name: string | null;
  company_business_no: string | null;
  status: string;
  file_key: string | null;
  reject_reason: string | null;
  reviewed_by: string | null;
  created_at: string;
  updated_at: string;
}

export interface VerificationListResponse {
  items: VerificationRead[];
  total: number;
}

File: backend/app/schemas/verification.py
from datetime import datetime

from pydantic import BaseModel, Field


class VerificationSubmit(BaseModel):
    file_key: str = Field(..., min_length=1, max_length=500)


class VerificationRead(BaseModel):
    id: str
    company_id: str
    company_name: str | None = None
    company_business_no: str | None = None
    status: str
    file_key: str | None = None
    reject_reason: str | None = None
    reviewed_by: str | None = None
    created_at: datetime
    updated_at: datetime


class VerificationListResponse(BaseModel):
    items: list[VerificationRead]
    total: int


class VerificationReview(BaseModel):
    status: str = Field(..., pattern=r"^(APPROVED|REJECTED)$")
    reject_reason: str | None = None
